### ISSUE TITLE
Convert Pytorch Densenet to caffe error

### ISSUE THREAD

**Issue by hiwudery** [hahnyuan/nn_tools#9]
Hi,
 I modify the example/vgg19_pytorch_to_caffe.py to convert densenet. The densenet is from torchvision, but the convert result is failed. Other models from torchvision are ok. Please help me to fix the problem of densenet.
Thank you
Jack

**Comment by hiwudery** [comment]
Hi:
I use the following code to convert densenet to caffe but failed.
Please help me to fix the problem.
Thank you
-Jack

```
import sys
sys.path.insert(0,'.')
sys.path.append('../')
import torch
from torch.autograd import Variable
from torchvision.models.densenet import *
import pytorch_to_caffe

if __name__=='__main__':
    name='densenet121'
    net=densenet121(True)
    input=Variable(torch.ones([1,3,224,224]))
    pytorch_to_caffe.trans_net(net,input,name)
    pytorch_to_caffe.save_prototxt('{}.prototxt'.format(name))
    pytorch_to_caffe.save_caffemodel('{}.caffemodel'.format(name))

**Comment by hahnyuan** [comment]
Hi

Thnaks for your remind.
I have figured out the problem! That is the new version of pytorch implemented the BatchNorm with num_batches_tracked added, which will +=1 at training phase.
The only thing you should do is add a `net.eval()` to solve this problem. I have commit a `example/densenet_pytorch_to_caffe.py`.
Sorry for reply so many days later because I use a older version of pytorch and cannot repeat your problem.

Zhiahng
